### PR TITLE
[SuperAdmin] Add optional word clause to query for providers

### PR DESCRIPTION
**Pull request by FaisalIqbal211**
## Describe your changes

https://www.loom.com/share/93cb4eba1c65413bace24e66984a6121

## Issue ticket number and link

Closes https://github.com/stakwork/sphinx-tribes/issues/1547

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist before requesting a review
- [x] I have performed a self-review of my code
- [x] I have tested on Chrome and Firefox
- [ ] I have tested on a mobile device
- [x] I have provided a screenshot or recording of changes in my PR if there were updates to the frontend

### CODE REVIEW

**Comment by FaisalIqbal211** [comment]
@ecurrencyhodler I have also added a query for the 'metrics/bounties/count' endpoint to ensure alignment with the "metrics/bounties" endpoints. Additionally, I have provided unit tests for both scenarios: metrics bounties and metrics bounties count.

File: db/metrics.go [hunk 1]
```diff
@@ -193,7 +193,14 @@ func (db database) GetBountiesByDateRange(r PaymentDateRange, re *http.Request)
 		limitQuery = fmt.Sprintf("LIMIT %d  OFFSET %d", limit, offset)
 	}
 
-	query := `SELECT * FROM public.bounty WHERE created >= '` + r.StartDate + `'  AND created <= '` + r.EndDate + `'`
+	var providerCondition string
```
Comment (kevkevinpal) [inline db/metrics.go:196]: let's rename this to `orgFilter` since it matches `org_uuid` better

File: handlers/metrics.go [hunk 2]
```diff
@@ -227,6 +233,8 @@ func (mh *metricHandler) MetricsBountiesCount(w http.ResponseWriter, r *http.Req
 		return
 	}
 
+	request.Providers = providers
```
Comment (elraphty) [inline handlers/metrics.go:236]: Don't pass the providers to the request variable, get it from the request.Query

File: db/structs.go [hunk 3]
```diff
@@ -611,6 +611,7 @@ type PaymentDateRange struct {
 	StartDate   string      `json:"start_date"`
 	EndDate     string      `json:"end_date"`
 	PaymentType PaymentType `json:"payment_type,omitempty"`
+	Providers   []string    `json:"providers,omitempty"`
```
Comment (kevkevinpal) [inline db/structs.go:614]: let's change this to `OrgFilter`

File: handlers/metrics_test.go [hunk 4]
```diff
@@ -173,6 +174,109 @@ func TestMetricsBounties(t *testing.T) {
...
+				ID:          6,
+				OwnerID:     "provider3",
+				Price:       200,
+				Title:       "bounty 6",
+				Description: "test bounty",
+				Created:     1113,
+			},
+		}
+		// Mock the database call to return bounties for the selected providers
+		mockDb.On("GetBountiesByDateRange", dateRange, req).Return(bounties).Once()
```
Comment (elraphty) [inline handlers/metrics_test.go:248]: You don't need to mock this amount of Providers, the you don't have this amount of DB calls in the request handler

**Comment by elraphty** [comment]
@FaisalIqbal211 The providers should be a comma separate string `{pubkey},{pubkey}` in the URL query, just like the format for languages.

File: db/metrics.go [hunk 5]
```diff
@@ -226,9 +233,16 @@ func (db database) GetBountiesByDateRangeCount(r PaymentDateRange, re *http.Requ
 		statusQuery = ""
 	}
 
+	var providerCondition string
+	if len(r.Providers) > 0 {
+		providerCondition = " AND owner_id IN ('" + strings.Join(r.Providers, "','") + "')"
```
[the remark this answers is not in the document]
Comment (FaisalIqbal211) [reply db/metrics.go:238]: @kevkevinpal provider is the person who created the bounty. Please check these comments https://github.com/stakwork/sphinx-tribes/issues/1547#issuecomment-1958095625 and https://github.com/stakwork/sphinx-tribes/issues/1547#issuecomment-1958130467

File: handlers/metrics.go (hunk 2 above)
Comment (FaisalIqbal211) [reply handlers/metrics.go]: Addressed

File: handlers/metrics_test.go (hunk 4 above)
Comment (FaisalIqbal211) [reply handlers/metrics_test.go:215]: Addressed

File: db/metrics.go [hunk 6]
```diff
@@ -193,7 +193,14 @@ func (db database) GetBountiesByDateRange(r PaymentDateRange, re *http.Request)
 		limitQuery = fmt.Sprintf("LIMIT %d  OFFSET %d", limit, offset)
 	}
 
-	query := `SELECT * FROM public.bounty WHERE created >= '` + r.StartDate + `'  AND created <= '` + r.EndDate + `'`
+	var providerCondition string
+	if len(r.Providers) > 0 {
+		providerCondition = " AND owner_id IN ('" + strings.Join(r.Providers, "','") + "')"
+	} else {
+		providerCondition = ""
```
[the remark this answers is not in the document]
Comment (FaisalIqbal211) [reply db/metrics.go]: Addressed

**Comment by FaisalIqbal211** [comment]
> @FaisalIqbal211 The providers should be a comma separate string `{pubkey},{pubkey}` in the URL query, just like the format for languages.

@elraphty Done

**Comment by FaisalIqbal211** [comment]
@elraphty @kevkevinpal Needs your attention at this PR.

File: handlers/metrics_test.go [hunk 7]
```diff
@@ -173,6 +174,64 @@ func TestMetricsBounties(t *testing.T) {
...
+			EndDate:   "2222",
+		}
+		body, _ := json.Marshal(dateRange)
+		req, err := http.NewRequestWithContext(ctx, http.MethodPost, "/boutnies", bytes.NewReader(body))
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		// Provide multiple provider IDs in the request query parameters
+		req.URL.RawQuery = "provider=provider1&provider=provider2&provider=provider3"
```
[the remark this answers is not in the document]
Comment (FaisalIqbal211) [reply handlers/metrics_test.go]: @elraphty Addressed

**Pull request merged**